src/TextField/TextField.jsx

```diff
@@ -351,7 +351,17 @@ const TextField = React.createClass({
   },
 
   _handleInputChange(e) {
-    this.setState({hasValue: isValid(e.target.value)});
+    let isAutofilled;
```

Comment:
Actually, what do you think about running this test only if `isValid(e.target.value)` is `false`?
We could save some CPU cycles :).